include/clap/ext/state-context.h

```diff
@@ -20,21 +21,27 @@
 ///        clap_plugin_state_context.save(CLAP_STATE_CONTEXT_FOR_PRESET),
 ///        CLAP_STATE_CONTEXT_FOR_PRESET)
 ///
```

Comment:
I suggest we remove the part starting with "All three operations should be equivalent". Instead, we state that hosts are required to use this extension for all save and load operations if the plugin supports the extension.
Mixing the two state extensions is confusing I think.

Reply:
That doesn't work for host released before that extension.

Reply:
It also makes perfect sense to keep this as "FOR PRESET" since the PROJECT context will allow a full association with bound hardware resources. I would keep both extensions, one is for simpler plugins/hosts and projects, the second one for the advanced ones.

Reply:
Sorry for my bad wording. My point is that a host should use just one of the state extensions when dealing with a particular plugin instance. Ideally it uses the new context one. If it does use the context extension it should use it for all saving and loading, rather than use the old one for some things and the new one for other things.